packages/admin/cms-admin/src/generator/future/utils/generateFormValuesTypeDefinition.ts

```diff
@@ -0,0 +1,54 @@
+import objectPath from "object-path";
+
+import { FormConfigInternal, FormFieldConfigInternal } from "../generator";
+import { convertObjectToStructuredString, FieldsObjectType } from "./convertObjectToStructuredString";
+import { RootBlocks } from "./findRootBlocks";
+import { getRootProps } from "./generateFieldList";
+
+export function generateFormValuesTypeDefinition({
+    config,
+    rootBlocks,
+    fragmentName,
+}: {
+    config: FormConfigInternal;
+    rootBlocks: RootBlocks;
+    fragmentName: string;
+}) {
+    const numberFields = config.fields.filter((field) => field.type == "number");
+    const rootPropsContainingNumberField = getRootProps(numberFields.map((field) => field.name));
```

Comment:
IMO a better name would be `numberFieldsRootProps`

Reply:
IMO `rootPropsContainingNumberField` allows less room for wrong interpretation, as `numberFieldsRootProps` could maybe mistaken as "root props with type number"